src/services/electrum/node.tsx

```diff
@@ -83,7 +84,13 @@ export default class Node {
   }
 
   public static getAllNodes(): NodeDetail[] {
-    return dbManager.getCollection(RealmSchema.NodeConnect) as unknown as NodeDetail[];
+    const { bitcoinNetworkType } = store.getState().settings;
+    return (dbManager.getCollection(RealmSchema.NodeConnect) as unknown as NodeDetail[]).filter(
+      (node) =>
+        bitcoinNetworkType === NetworkType.TESTNET
```

Comment:
I don't think this is a reliable way to know the node network, some testnet nodes may not have testnet in the URL.